### ISSUE TITLE
How to manage multiple vitepress website with azure static web app using base url ?

### ISSUE THREAD

**Issue by kai-you2** [vuejs/vitepress#2386]
## Overview 

We have multiple service we want to have documentation using vitepress, I know we can easily deploy the website using static web app, **but it seems impossible to host those documentation site under one domain with base url ?**

What I need is something like  
- https://companydomain/web/projectA -> projectA's vitepress website 
- https://compnaydomain/web/projectB -> projectB's vitepress website 

Is that possible using azure static web app? If not, I am considering using azure app service, since it provide virtual application which may seems to fit in these kind of situation, but in this case, how do we host a website in azure web app with vitepress ? since it didn't have a nodjs server

**Comment by brc-dd** [comment]
> but it seems impossible to host those documentation site under one domain with base url

What's the issue you're encountering? It should be straightforward to build those projects and deploy them 👀 (azure static web apps should work fine IMO)

**Comment by kai-you2** [comment]
Hi, thanks for taking interesting answering the problem, I posted a more detailed article at https://stackoverflow.com/questions/76050387/how-to-deploy-my-vitepress-to-azure-static-web-app-with-base-url

Majorly the problem would be how to configure base url to static web app, I tried to set the outDir for the projects to be `/web/ui` for example, but in my github workflow, I still need to set the `app_location` to be `.vitepress/dist/web/ui` in order to find the `index.html`, which may it to be pointless since it's still treat `/` as it's root url, I want to be able to point one domain to multiple vitepress projects with different base url for my client

ex: http://companydomain/web/ui/ -->  A project
  http://companydomain/web/sdk/ -->  B project
http://companydomain/web/doc/ -->  C project

**Comment by brc-dd** [comment]
Ah, I'll try to share an example. It's might be hard to explain this way.

The basic approach is:

- project A has base `/web/ui/` --> set its outDir to `../dist/web/ui`
- project B has base `/web/sdk/` --> set its outDir to `../dist/web/sdk`
- same for project C

After you've build all three projects, you'll have a dist directory like this:

```ts

dist/web/
├─ ui/
├─ sdk/
├─ doc/
```

You need to upload that to Azure.

**Comment by kai-you2** [comment]
Hi, thanks for your answer. But if I upload the `dist`, the azure static web app will be looking for index.html in `dist` level, which will not exist, which cause the build pipeline to fail.

**Comment by brc-dd** [comment]
You can emit a blank html file to dist or have some project with base `/` too. Because anyway, those three projects won't be accessible from `https://companydomain/` or `https://companydomain/web`

**Comment by kai-you2** [comment]
I see, is there a way for all three projects to be configuring at three different static web app then using the same domain to point to those web apps ?  Seems these three projects are located at three different repo, so it seems impossible to do it in your structures.

**Comment by brc-dd** [comment]
I don't think that will be possible. Route rewrites could've been useful, but IG Azure needs them relative to your app root, which will be different for all three projects. If you put something like Cloudflare in front of Azure, then it might be possible to configure rewrites there and then use. Things like Azure NGINXaaS might also work, but I haven't used them.

I'd recommend asking a question on StackOverflow (Azure Collective) stating that you've three different apps, you want to serve them under the same domain but different sub directories.

**Comment by kai-you2** [comment]
Thanks for your answer, acutally I found a different approach using `azure blob storage` using static web app which you can just upload different `dist` to different sub folder, I am not sure if that's a great way for deploying `vitepress` , but for now it seems working, my best choice still would be azure static web app, since it comes with more advantage. but thanks again for helping!

**Comment by brc-dd** [comment]
Azure blob might have slightly degraded performance, because they are not exactly meant for serving sites. IIRC blob was more like s3 than cloudfront. Azure CDN endpoints might give significantly better performance here.

Anyway, there are a bunch of similar threads of StackOverflow, where people have recommended using Azure Application Gateway and Virtual Directories using which you can have multiple apps under the same domain.

Some threads for reference:
- https://stackoverflow.com/q/73127391
- https://stackoverflow.com/q/65200778
- https://stackoverflow.com/q/55624386
- https://stackoverflow.com/q/46033446
- https://stackoverflow.com/q/56452192
